pkgs/top-level/static.nix

```diff
@@ -151,4 +155,19 @@ in {
     };
   };
 
+  aws-sdk-cpp = super.aws-sdk-cpp.override {
+    staticOnly = true;
+  };
+
+  brotli = super.brotli.override {
+    staticOnly = true;
+  };
+
+  curl = super.curl.override {
+    gssSupport = false;
+  };
```

Comment:
I would imagine your target audience for this is users who are running and deploying on rootless "enterprise" environments such a RHEL or CentOS.  It's worth noting that in many of these setups, kerberos is used extensively for all HTTP and git authentication, so if you could get it working in this static variant it'd probably be worthwhile.

Reply:
Yeah this is an issue. libkrb5 doesn't have good way to link statically, however:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=439039

kerberos heimdal might support it though.